localstack-core/localstack/services/events/provider.py

```diff
@@ -243,7 +250,9 @@ def describe_event_bus(
         self, context: RequestContext, name: EventBusNameOrArn = None, **kwargs
     ) -> DescribeEventBusResponse:
         name = extract_event_bus_name(name)
-        store = self.get_store(context)
+        region = context.region
```

Comment:
In case we get ARN, I believe we should use it for extracting region and account_id

Reply:
I manually checked and so far I had no luck using describe for an event bus in a different region / account against AWS. I believe that this behavior is only possible for `put_events` but I will add a specific test for each API endpoint that allows for event bus ARN. 
This is also why I only pars the region and account_id for `put_events` since cross region / account is only valid for this operation.

Reply:
AWS has the following behavior that I could confirm against AWS for `put_events`:
- cross-account is possible with the correct permissions for custom event busses
- cross-region is not allowed for custom event busses with a client-side error
- cross-account and cross-region does not work for default event busses but there is no error message

Since this is a much larger endeavor to test all API endpoints for cross-region and cross-account, as well as add error messages for not allowing cross-region for custom event buses and not allowing region as well as cross-account for default bus but without raising an error to be fully compatible with AWS behavior - I will deal with this in a followup pr.
This PR is purely concerned with fixing the cross-account and region event bus to event bus communication for allowing IAM policy enforcement.